core/Cargo.toml

```diff
@@ -189,7 +189,8 @@ wasmtimer = { version = "0.2.0", default-features = false, features = [
 ws_stream_wasm = "0.7.4"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-tokio = { version = "1.40.0", default-features = false, features = [
+easy-parallel = "3.3.1"
```

Comment:
With this patch the test block indefinitely, as you can see with this test:

    cargo test  -F kv-mem --package surrealdb --test planner select_memory_ordered_collector

The reason is that the group of threads requires the following future to be executed:

```Rust
futures::join!(tasks.0, tasks.1, tasks.2, tasks.3)
```

Otherwise, it is indefinitely waiting for a message to be sent. But no message will be sent until these tasks are running.

That is where the 'finish' method of EasyParallel was useful.

https://github.com/smol-rs/easy-parallel/blob/ac2652b42ad183c819ae0025b2db2cfb88072ca8/src/lib.rs#L258C1-L258C52

Any idea what is the best way to achieve that with the new thread scope logic?

Reply:
Oh, misread where the call to the closure is, if you move 
```rust
block_on(async {
			// Wait for all closures
			futures::join!(tasks.0, tasks.1, tasks.2, tasks.3);
			// Stop every threads
			drop(signal);
		});
```
To before where the handles are joined.
It should be equivalent to the easy_parallel implementation.

Reply:
It is! Thanks a lot!